File: code_samples/csharp/testing_assertions.cs
using System;

/// <summary>
/// Testing: Assertions
/// AI/ML Training Sample
/// </summary>
public class Assertions
{
    private string data;
    
    public Assertions()
    {
        this.data = string.Empty;
    }
    
    public void Process(string input)
    {
        this.data = input;
    }
    
    public string GetData()
    {
        return this.data;
    }
    
    public bool Validate()
    {
        return !string.IsNullOrEmpty(this.data);
    }
    
    public static void Main(string[] args)
    {
        var instance = new Assertions();
        instance.Process("example");
        Console.WriteLine($"Data: {instance.GetData()}");
        Console.WriteLine($"Valid: {instance.Validate()}");
    }
}

File: code_samples/go/testing_assertions.go
package main

import (
    "fmt"
)

// Testing: Assertions
// AI/ML Training Sample

type Assertions struct {
    Data string
}

func NewAssertions() *Assertions {
    return &Assertions{
        Data: "",
    }
}

func (s *Assertions) Process(input string) {
    s.Data = input
}

func (s *Assertions) Validate() bool {
    return len(s.Data) > 0
}

func (s *Assertions) GetData() string {
    return s.Data
}

func main() {
    instance := NewAssertions()
    instance.Process("example")
    fmt.Printf("Data: %s\n", instance.GetData())
    fmt.Printf("Valid: %v\n", instance.Validate())
}
